ext/AvizExt/viz/rule_extraction.jl

```diff
@@ -0,0 +1,114 @@
+using SIRUS
+import ADRIA.analysis: Rule
+
+"""
+    ADRIA.viz.rules_scatter(rs::ResultSet, scenarios::DataFrame, clusters::Vector{Int64}, outcomes::NamedDimsArray, rules::Vector{Rule{Vector{Vector},Vector{Float64}}}; fig_opts::Dict=Dict(), axis_opts::Dict=Dict())
+
+# Description
+For each condition Rule, plots a scatter showing one condition clause at each axis.
+The target area of each condition is algo highlited, and the positive and negative
+class data have different colors. This maybe be usefull for bumphunting.
+
+# Arguments
+- `rs` : ResultSet
+- `scenarios` : Scenarios used to generate ResultSet
+- `clusters` : Clusters used to extract Rule using SIRUS
+- `outcomes` : Results of scenario metric
+- `rules` : Rules extracted from scenarios and clusters
+
+# Returns
+Figure with condition Rule scatter plots
+"""
+function ADRIA.viz.rules_scatter(rs::ResultSet, scenarios::DataFrame, clusters::Vector{Int64}, outcomes::NamedDimsArray, rules::Vector{Rule{Vector{Vector},Vector{Float64}}}; fig_opts::Dict=Dict(), axis_opts::Dict=Dict())
+    f = Figure(; fig_opts...)
+    g = f[1, 1] = GridLayout()
+
+    # For now we are only plotting conditions with two clauses
+    rules = filter(r -> length(r.condition) == 2, rules)
+
+    ADRIA.viz.rules_scatter!(g, rs, scenarios, clusters, outcomes, rules; axis_opts=axis_opts)
+
+    return f
+end
+function ADRIA.viz.rules_scatter!(g::Union{GridLayout,GridPosition}, rs::ResultSet, scenarios::DataFrame, clusters::Vector{Int64}, outcomes::NamedDimsArray, rules::Vector{Rule{Vector{Vector},Vector{Float64}}}; axis_opts::Dict=Dict())
+    sub_g = g[1, 1] = GridLayout()
+
+    # Target cluster index
+    target_index = clusters .== ADRIA.analysis.target_cluster(clusters, outcomes)
+
+    # To get human readable names later
+    ms = model_spec(rs)
+
+    n_factors = length(rules)
+    n_rows, n_cols = _calc_gridsize(n_factors)
+
+    for r in 1:n_rows
+        for c in 1:n_cols
+            # Get condition clauses to be shown in x and y axis
+            index = c + (r - 1) * n_cols
+            if length(rules) < index
+                break
+            end
+            condition = rules[index].condition
+
+            # Feature names
+            condition_features = first.(condition)
+            foi = ms.fieldname .∈ [condition_features]
+            h_names = ms[foi, :name]
```

Comment:
I think this extracts the factors/features of interest but does not check for order so it is possible for the figure axis labels to be in the wrong order